q2_longitudinal/_utilities.py

```diff
@@ -404,130 +402,6 @@ def _regplot_subplots_from_dataframe(state_column, metric, metadata,
-                            plot_control_limits=True, xtick_interval=None,
-                            yscale='linear', spaghetti='no'):
-
-    groups = metadata[group_column].unique()
-    states = sorted(metadata[state_column].unique())
-    fig_count = len(groups) + 1
-    chart, axes = plt.subplots(fig_count, figsize=(6, fig_count * 6))
-
-    # determine x tick interval: autoscale so that ≤ 30 labels appear
-    xtick_interval = _set_xtick_interval(xtick_interval, states)
-
-    # plot individual groups' control charts
-    cmap = dict(zip(groups, sns.color_palette(palette, n_colors=len(groups))))
-    for num, (group, group_md) in enumerate(metadata.groupby(group_column), 1):
-        c, gm, gs = _control_chart(
-            state_column, metric, group_md, None, ci=ci, legend=False,
-            color=cmap[group], plot_control_limits=plot_control_limits,
-            ax=axes[num], palette=None, xtick_interval=xtick_interval)
-        c.set_title('{0}: {1}'.format(group_column, group))
-        if spaghetti != 'no':
-            # plot group's sphaghetti on main plot and current subplot
-            for ax in [0, num]:
-                c = _make_spaghetti(
-                    group_md, state_column, metric, individual_id_column,
-                    states, ax=axes[ax], color=cmap[group], alpha=0.3,
-                    spaghetti=spaghetti)
-        c = _set_xticks(c, group_md, state_column, states, xtick_interval)
-        axes[num].set_yscale(yscale)
-        num += 1
-
-    # plot all groups together, compare variances
-    c, global_mean, global_std = _control_chart(
-        state_column, metric, metadata, group_column, ci=ci, palette=cmap,
-        plot_control_limits=plot_control_limits, ax=axes[0],
-        xtick_interval=xtick_interval)
-    c.set_title('Group volatility comparison plot')
-    c = _set_xticks(c, metadata, state_column, states, xtick_interval)
-
-    c.set_yscale(yscale)
-
-    plt.tight_layout()
-
-    return chart, global_mean, global_std
-
-
-def _make_spaghetti(metadata, state_column, metric, individual_id_column,
-                    states, ax, color=None, alpha=1.0, spaghetti='no'):
-    for ind, ind_data in metadata.groupby(individual_id_column):
-        # optionally plot mean of replicates
-        if spaghetti == 'mean':
-            ind_data = ind_data.groupby(state_column).mean()
-            ind_data[state_column] = ind_data.index
-        altered_states = ind_data[state_column]
-        # Adjust xticks so that it follows a pseudo-categorical scale
-        # (e.g., 0, 1, 7, 200 would be plotted at even intervals on x axis)
-        # so that spaghetti aligns with seaborn pointplot x axis.
-        if states is not None:
-            altered_states = altered_states.apply(states.index)
-
-        ax.plot(altered_states, ind_data[metric], alpha=alpha, c=color,
-                label='_nolegend_')
-    return ax
-
-
-def _set_xtick_interval(xtick_interval, states):
```

Comment:
This is not a feature that we want to strip away (though there must be a better way to do this in vega rather than keeping this function). 

See the broken qzv I've attached to the review to see why... we either need to support explicit interval setting or add sensible auto-scaling in vega.